cmd/clusterctl/client/tree/discovery.go

```diff
@@ -36,8 +40,10 @@ type DiscoverOptions struct {
 	// ShowMachineSets instructs the discovery process to include machine sets in the ObjectTree.
 	ShowMachineSets bool
 
+	// ShowAllResources instructs the discovery process to include all resources associated with the cluster.
+	ShowAllResources bool
```

Comment:
I felt that grouping the CRDs in a tree was helpful for me to understand how the cluster is logically structured, and how the resources relate to each other. I thought that for a beginner, it could help show how all the resources are being used. 

As we discussed earlier, this flag would also be useful for discovering all the CRDs and returning them in a tree for the CAPI GUI prototype. I see what you mean about how the output could become very large for clusters with many machines, and I'd like these changes to improve the client for both purposes. I'm open to design suggestions as well, do you think it'd be better to remove `ShowAllResources` as a CLI flag and leave it in `DiscoverOptions`?

Reply:
Let's start by dropping from the new flags from the CLI; also, in DiscoverOptions let's add a warning about usage with Cluster with a high number of machines

Reply:
Also, for sake of clarity, I suggest breaking down `ShowAllResources` into `ShowCustomResourceSets` and `ShowTemplates`; by implementing the fundamental capabilities first, we make it possible to add the ShowAll flag later or to combine them in different ways.